app/(main)/settings/components/card.tsx

```diff
@@ -4,15 +4,22 @@ import type { FC, PropsWithChildren } from "react";
 
 type SubmitAction = {
 	href?: never;
+	component?: never;
 	content: string;
 	onAction: () => void;
 };
 type LinkAction = {
 	onAction?: never;
+	component?: never;
 	content: string;
 	href: string;
 };
-type Action = SubmitAction | LinkAction;
+type Comp = {
+	component: React.ReactNode;
+	onAction?: never;
+	href?: never;
+};
+type Action = SubmitAction | LinkAction | Comp;
```

Comment:
I want the name to be consistent.

```suggestion
type CustomComponentAction = {
	component: React.ReactNode;
	onAction?: never;
	href?: never;
};
type Action = SubmitAction | LinkAction | CustomComponentAction;
```